forem/forem
# Create feed impression and click events

## What type of PR is this? (check all applicable)


## Description

This one is a bit large, but lays a bit of a foundation for follow-up reaction and comment events. I have also split off bumping event counters and periodically compacting feed events to a different branch

## Related Tickets & Documents



- Related Issue #19924
- Closes #19924 

## QA Instructions, Screenshots, Recordings

https://www.loom.com/share/b4847236745b496ba3bf3c44726b8076

I did not mention in the video that if you scroll back and forth you should see that new events don't get sent from the same feed/tab. Also, if you open a few tabs and scroll within them at the same time (logged in to the same account) you should see that only one impression event per article gets recorded (as documented this is time-windowed, and will only prevent duplicates within the previous five minutes to reduce spam and noise)

### UI accessibility checklist

Made UI changes but they are background behaviour changes only.

## Added/updated tests?


## [optional] Are there any post deployment tasks we need to perform?

Nope

## [optional] What gif best describes this PR or how it makes you feel?

![tired-exhausted](https://github.com/forem/forem/assets/29291843/fba28607-e297-4746-a5b0-762af7cf5ff9)
